homeassistant/components/omnilogic/sensor.py

```diff
@@ -0,0 +1,576 @@
+                for bow in backyard["BOWS"]:
+                    if bow["systemId"] == self._system_id:
+                        sensor_data = bow["waterTemp"]
+                        break
+
+        elif self._kind == "air_temperature":
+            for backyard in self.coordinator.data:
+                if backyard.get("systemId") == self._system_id:
+                    sensor_data = backyard.get("airTemp")
+
+        temp_return = int(sensor_data)
+        temp_state = int(sensor_data)
+        unit_of_measurement = TEMP_FAHRENHEIT
+        if self._backyard["Unit-of-Measurement"] == "Metric":
+            temp_return = round((temp_return - 32) * 5 / 9, 1)
+            unit_of_measurement = TEMP_CELSIUS
+
+        if int(sensor_data) == -1:
+            temp_return = None
+            temp_state = None
+
+        self._attrs["hayward_temperature"] = temp_return
+        self._attrs["hayward_unit_of_measure"] = unit_of_measurement
+        if temp_state is not None:
+            self._state = float(temp_state)
+            self._unit = TEMP_FAHRENHEIT
+
+        return self._state
+
+
+class OmniLogicPumpSpeedSensor(OmnilogicSensor):
+    """Define an OmniLogic Pump Speed Sensor."""
+
+    def __init__(
+        self,
+        coordinator: OmniLogicUpdateCoordinator,
+        kind: str,
+        name: str,
+        backyard: dict,
+        bow: dict,
+        device_class: str,
+        icon: str,
+        unit: str,
+        sensor_data: dict,
+    ):
+        """Initialize the sensor."""
+
+        name = f"{backyard['BackyardName']} {bow['Name']} {name}"
+
+        super().__init__(
+            coordinator=coordinator,
+            kind=kind,
+            name=name,
+            backyard=backyard,
+            bow=bow,
+            device_class=device_class,
+            icon=icon,
+            unit=unit,
+            sensor_data=sensor_data,
+            system_id=sensor_data.get("systemId"),
+        )
+
+    @property
+    def state(self):
+        """Return the state for the pump speed sensor."""
+
+        sensor_data = {}
+        pump_type = ""
+        pump_speed = 0
+
+        if self._kind == "filter_pump_speed":
+            for backyard in self.coordinator.data:
+                for bow in backyard.get("BOWS"):
+                    if bow["Filter"]["systemId"] == self._system_id:
+                        sensor_data = bow["Filter"]
+                        break
+
+            pump_type = PUMP_TYPES.get(sensor_data.get("Filter-Type"))
+            pump_speed = sensor_data.get("filterSpeed")
+
+        elif self._kind == "pump_speed":
+            for backyard in self.coordinator.data:
+                for bow in backyard.get("BOWS"):
+                    for pump in bow.get("Pumps"):
+                        if pump["systemId"] == self.entity_data["systemId"]:
+                            sensor_data = pump
+                            break
+
+            if sensor_data.get("Filter-Type") == "PMP_VARIABLE_SPEED_PUMP":
```

Comment:
We can use the constant dict here too to look up pump type.

Reply:
Good call - have addressed and added the pump types to the filter pump types in the constant dict.